fusedrug_examples/interaction/drug_target/affinity_prediction/t5_mol/tokenizer/multi_tokenizer.py

```diff
@@ -0,0 +1,1304 @@
+
+                - If ``is_pretokenized=False``: :class:`~tokenizers.TextEncodeInput`
+                - If ``is_pretokenized=True``: :class:`~tokenizers.PreTokenizedEncodeInput`
+
+            is_pretokenized (:obj:`bool`, defaults to :obj:`False`):
+                Whether the input is already pre-tokenized
+
+            add_special_tokens (:obj:`bool`, defaults to :obj:`True`):
+                Whether to add the special tokens
+
+        Returns:
+            A :obj:`List` of :class:`~tokenizers.Encoding`: The encoded batch
+
+        """
+        raise Exception("Not implemented")
+
+    @staticmethod
+    def from_buffer(buffer: object) -> object:
+        """
+        Instantiate a new :class:`~tokenizers.Tokenizer` from the given buffer.
+
+        Args:
+            buffer (:obj:`bytes`):
+                A buffer containing a previously serialized :class:`~tokenizers.Tokenizer`
+
+        Returns:
+            :class:`~tokenizers.Tokenizer`: The new tokenizer
+        """
+        raise Exception("Not implemented")
+
+    @staticmethod
+    def from_file(path: str) -> object:
+        """
+        Instantiate a new :class:`~tokenizers.Tokenizer` from the file at the given path.
+
+        Args:
+            path (:obj:`str`):
+                A path to a local JSON file representing a previously serialized
+                :class:`~tokenizers.Tokenizer`
+
+        Returns:
+            :class:`~tokenizers.Tokenizer`: The new tokenizer
+        """
+        raise Exception("Not implemented")
+
+    @staticmethod
+    def from_pretrained(
+        identifier: str,
+        revision: Optional[str] = "main",
+        auth_token: Optional[str] = None,
+    ) -> Any:
+        """
+        Instantiate a new :class:`~tokenizers.Tokenizer` from an existing file on the
+        Hugging Face Hub.
+
+        Args:
+            identifier (:obj:`str`):
+                The identifier of a Model on the Hugging Face Hub, that contains
+                a tokenizer.json file
+            revision (:obj:`str`, defaults to `main`):
+                A branch or commit id
+            auth_token (:obj:`str`, `optional`, defaults to `None`):
+                An optional auth token used to access private repositories on the
+                Hugging Face Hub
+
+        Returns:
+            :class:`~tokenizers.Tokenizer`: The new tokenizer
+        """
+        raise Exception("Not implemented")
+
+    @staticmethod
+    def from_str(json: str) -> object:
+        """
+        Instantiate a new :class:`~tokenizers.Tokenizer` from the given JSON string.
+
+        Args:
+            json (:obj:`str`):
+                A valid JSON string representing a previously serialized
+                :class:`~tokenizers.Tokenizer`
+
+        Returns:
+            :class:`~tokenizers.Tokenizer`: The new tokenizer
+        """
+        raise Exception("Not implemented")
+
+    def get_vocab(self, with_added_tokens: Optional[bool] = True) -> Dict:
```

Comment:
Don't we have it implemented already for the decode part?

Reply:
Do you mean get_vocab? Not sure what we want it to do. In the original tokenizer it returns a dict of {token:id}. In modular tokenizer, the same token may map to different IDs, depending on context.

Reply:
It may be possible to just return a set of tokens, although I'm not sure what will be the use of that, plus, it will change the signature of the method. I suggest we implement this once we have a clear use case.